xdsl/xdsl_opt_main.py

```diff
@@ -299,18 +313,19 @@ def parse_input(self) -> ModuleOp:
             _, file_extension = os.path.splitext(self.args.input_file)
             file_extension = file_extension.replace(".", "")
 
+        chunks = [f]
+        if self.args.split_input_file:
+            chunks = list(map(StringIO, f.read().split("// -----")))
```

Comment:
nit:
```suggestion
            chunks = [StringIO(chunk) for chunk in f.read().split("// -----")]
```
List comprehensions are more readable and slightly faster